tests/test_karma_chameleon.py

```diff
@@ -110,13 +110,24 @@ def test_increment(self, app_inc_karma) -> None:
     @mock.patch("karma_chameleon.bot.KarmaBot.decrement_karma")
     def test_decrement(self, app_dec_karma) -> None:
         """Test the method of the main app which calls the bot decrement."""
-        msg = "Got the message!"
+        for msg in ["Got the message!", ""]:
+            app_dec_karma.return_value = msg
+
+            with contextlib.redirect_stdout(io.StringIO()) as out:
+                decrement(self.test_msg, self._say_method)
+                assert app_dec_karma.inc_karma.called_with(self.test_msg)
+                assert out.getvalue() == (msg + "\n" if msg else msg)
+
+    @mock.patch("karma_chameleon.bot.KarmaBot.decrement_karma")
+    def test_decrement_no_msg(self, app_dec_karma) -> None:
```

Comment:
Doesn't the test above do the same thing?

Reply:
Yes but in a different place, so not really lol.  In the version in test_karma_chameleon we mock out  the underlying logic so the bot part doesn't actually run.  This version just simulates the output from the bot and makes sure that the main app does the right thing.